python/podio_class_generator.py

```diff
@@ -205,6 +206,9 @@ def _process_datatype(self, name, definition):
     self._fill_templates('Obj', datatype)
     self._fill_templates('Collection', datatype)
 
+    if self.reader.options["createSIOHandlers"]:
```

Comment:
```suggestion
    if 'SIO' in self.reader.options["IOHandlers"]:
```
Or something like this, and making IOHandlers a list of IOHandlers to create (ROOT and SIO for the moment)

Reply:
That is a good idea. It would definitely scale better than the current approach in the yaml file of the datamodel. I will make this change, and since ROOT doesn't need any special treatment at the moment it will basically be simply ignored. Do you think that is a reasonable approach, or should we do something more explicit for IO systems that do not need additional generated code? E.g. only list the handlers that actually need additional code generation?

Reply:
After reading more of the PR.

Maybe this should be coupled to the CMake option for enabling SIO, in the base CMakeLists in the block
```cmake
if(SIO_HANDLERS)
  LIST(APPEND PODIO_IO_HANDLERS SIO)
endif()
```

The PODIO_GENERATE_DATAMODEL function could just set environment variables for each handler,
```cmake
FOREACH(HANDLER PODIO_IO_HANDLERS)
  SET(EXPORT_HANDLER "${EXPORT_HANDLER} ${HANDLER}_HANDLER=1")
ENDFOREACH()
```
This

https://github.com/AIDASoft/podio/blob/d4a02651332e2fe3ffbac4e99fa61b34600a16fa/cmake/podioMacros.cmake#L141

becomes
```
    COMMAND ${EXPORT_HANDLER} python ${podio_PYTHON_DIR}/podio_class_generator.py --quiet ${YAML_FILE} ${ARG_OUTPUT_FOLDER} ${datamodel}
```
and then the check becomes
```python
if os.environ.get('SIO_HANDLER'):
```

The variable  PODIO_IO_HANDLERS is exported to the podioConfig, so edm4hep can pick this up as well when calling PODIO_GENERATE_DATAMODEL

Which reminds me that the dependency on SIO should also be exported in the podioConfig, if enabled.

Reply:
> ROOT doesn't need any special treatment 

Actually, I just realized that this is not true, because the `selection.xml` file is only necessary for the ROOT dictionaries.

-----------
Regarding the handling of the generation of different IO systems, I have started some kind of discussion in that direction in #131. Maybe instead of setting environment variables, we could pass command line arguments to the generator from cmake? Or would that defeat the purpose of exporting `PODIO_IO_HANDLERS`? I, personally, find explicitly passing command line arguments a bit easier to debug and to follow the flow of the whole thing.
 
For exporting the dependency on SIO, it would be ideal if SIO already exports targets, right?